libraries/ui/src/SlideList.tsx

```diff
@@ -1,230 +1,231 @@
-import React, { useState } from 'react';
+import React, {
+  useState, useEffect, useRef, useCallback,
+} from 'react';
 import clsx from 'clsx';
+import { type SectionHeadingProps, SectionHeading } from './Section';
 
 export type SlideListProps = {
   className?: string;
-  title?: string;
-  subtitle?: string;
-  description?: string;
   children: React.ReactNode;
   featuredSlot?: React.ReactNode;
-  itemsPerSlide?: number;
-  slideClassName?: string;
-  containerClassName?: string;
-  slidesWrapperWidth?: string | { mobile: string; desktop: string };
-};
+  maxItemsPerSlide?: number;
+  minItemWidth?: number;
+} & SectionHeadingProps;
 
 export const SlideList: React.FC<SlideListProps> = ({
   className,
-  title,
-  subtitle,
-  description,
   children,
   featuredSlot,
-  itemsPerSlide = 1,
-  slideClassName,
-  containerClassName,
-  slidesWrapperWidth = { mobile: '100%', desktop: '800px' },
+  maxItemsPerSlide = 1,
+  minItemWidth = 260,
+  title,
+  titleLevel,
+  subtitle,
+  subtitleLevel,
```

Comment:
This component has ended up very complex unfortunately

Reply:
I think this could do with a simplification refactor, but it's already a big improvement